build.sbt

```diff
@@ -1420,7 +1420,6 @@ lazy val `directory-watcher-wrapper` = project
         scalaLibrary ++
         Seq(
           "org.slf4j"        % "slf4j-api" % "1.7.36",
-          "net.java.dev.jna" % "jna"       % jnaVersion
```

Comment:
- I have some mental problem with `filterModulesFromClasspath`
- it cannot grasp how it works
- it always seems _upside/down_
- why don't we just exclude the JNA dependency when declaring it, @Akirathan?

```
"io.methvin"       % "directory-watcher" % directoryWatcherVersion exclude("net.java.dev.jna", "jna"),
```

Reply:
In this case, excluding the dependency directly in `libraryDependencies` by using the code you wrote should do exactly the same thing. In general, that may not be the case. The core idea of `filterModulesFromClasspath` is simple: As input, it gets a collection of `ModuleID` (which is sbt's internal representation of module tripples, like `"io.methvin" % "directory-watcher" % "1.0.0"`) and as output, it returns list of files on filesystem that correspond to these modules. The motivation for these functions was that there is no way how to access the Ivy/Maven module resolution from sbt directly.